src/main/java/org/broadinstitute/hellbender/tools/copynumber/PostprocessGermlineCNVCalls.java

```diff
@@ -1,221 +1,593 @@
+        Utils.validateArg(unsortedIntervalCollectionsFromCalls.stream()
+                        .map(SimpleIntervalCollection::getMetadata)
+                        .map(LocatableMetadata::getSequenceDictionary)
+                        .allMatch(shardSAMSequenceDictionary ->
+                                shardSAMSequenceDictionary.equals(samSequenceDictionary)),
+                "The SAM sequence dictionary is not the same for all of the call shards.");
 
-        if (firstCopyNumberPosteriorFileInfo.getMiddle().size() < 3) {
-            throw new UserException.BadInput("There should be at least 3 copy number states.");
+        /* get the correct shard short orders and sort all collections */
+        final int[] sortedCallShardsOrder = SimpleIntervalCollection.getSimpleIntervalCollectionSortedOrder(
+                unsortedIntervalCollectionsFromCalls, samSequenceDictionary, true);
+        sortedCallDirectories = Arrays.stream(sortedCallShardsOrder)
+                .mapToObj(unsortedCallDirectories::get)
+                .collect(Collectors.toList());
+        final List<SimpleIntervalCollection> sortedIntervalCollectionsFromCalls = Arrays.stream(sortedCallShardsOrder)
+                .mapToObj(unsortedIntervalCollectionsFromCalls::get)
+                .collect(Collectors.toList());
+        sortedIntervalCollections = sortedIntervalCollectionsFromCalls;
+
+        /* assert that allosomal contigs are contained in the SAM sequence dictionary */
+        final Set<String> allContigs = samSequenceDictionary.getSequences().stream()
+                .map(SAMSequenceRecord::getSequenceName)
+                .collect(Collectors.toSet());
+        allosomalContigsSet = new HashSet<>(allosomalContigsList);
+        if (allosomalContigsSet.isEmpty()) {
+            logger.warn(String.format("Allosomal contigs were not specified; setting ref copy-number allele " +
+                    "to (%d) for all intervals.", refAutosomalCopyNumber));
+        } else {
+            Utils.validateArg(allContigs.containsAll(allosomalContigsSet), String.format(
+                    "The specified allosomal contigs must be contained in the SAM sequence dictionary of the " +
+                            "call-set (specified allosomal contigs: %s, all contigs: %s)",
+                    allosomalContigsSet.stream().collect(Collectors.joining(", ", "[", "]")),
+                    allContigs.stream().collect(Collectors.joining(", ", "[", "]"))));
         }
 
-        final VariantContextWriter outputWriter = createVCFWriter(outputFile);
-        final GermlineCNVPostprocessingEngine germlineCNVPostprocessingEngine = new GermlineCNVPostprocessingEngine(
-                outputWriter, copyNumberStateCollection, sampleName, samSequenceDictionary);
+        /* get sample name from the first shard and assert that all shards have the same sample name */
+        sampleName = getShardSampleName(0);
```

Comment:
The idea is that the sample name for each `call_shard_X/SAMPLE_Y` path is first and foremost specified by the `sample_name.txt`. The sample name found in the SAM header tag for each of the contained metadata files (which are many) must then match that in `sample_name.txt`. This may be redundant, though, is a kind of integrity check.